### ISSUE TITLE
[enhancement] Cautious Optimizers

### ISSUE THREAD

**Issue by kallewoof** [tdrussell/qlora-pipe#59]
https://github.com/kyleliang919/C-Optim

This looks really cool. I wonder how hard this would be to get working, considering the authors claim it's a single line code change.

**Comment by tdrussell** [comment]
If you want to try it out, you could just copy their optimizer implementation to a file and add an option for it in train.py. It looks like a normal pytorch optimizer subclass so it should just work.

**Comment by kallewoof** [comment]
You're probably right. I did some experimenting and it looks like it works, although I'm not sure if it's an improvement over optimi adamw_kahn. @jukofyork this thing pokes at weight decay, one of your favorite topics. Maybe of interest.

**Comment by kallewoof** [comment]
Will push a PR if this turns out to be useful.
